src/snowflake/snowpark/_internal/analyzer/schema_utils.py

```diff
@@ -130,7 +130,7 @@ def get_new_description(
 
 def run_new_describe(
     cursor: SnowflakeCursor, query: str
-) -> Union[List[ResultMetadata], List["ResultMetadataV2"]]:  # pyright: ignore
```

Comment:
The type hint here is correct, I opened a PR to fix the type hints in Python connector: https://github.com/snowflakedb/snowflake-connector-python/pull/1829